src/ResourceManagement/CosmosDB/SqlDatabaseImpl.cs

```diff
@@ -1,46 +1,248 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
+
 namespace Microsoft.Azure.Management.CosmosDB.Fluent
 {
+    using Microsoft.Azure.Management.CosmosDB.Fluent.Models;
     using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
 
     internal partial class SqlDatabaseImpl :
-        Wrapper<Models.SqlDatabaseGetResultsInner>,
-        ISqlDatabase
+        ExternalChildResource<ISqlDatabase,
+            SqlDatabaseGetResultsInner,
+            ICosmosDBAccount,
+            CosmosDBAccountImpl>,
+        ISqlDatabase,
+        SqlDatabase.Definition.IDefinition<CosmosDBAccount.Definition.IWithCreate>,
+        SqlDatabase.Update.IUpdate,
+        SqlDatabase.Definition.IDefinition<CosmosDBAccount.Update.IWithOptionals>
     {
-        internal SqlDatabaseImpl(Models.SqlDatabaseGetResultsInner innerObject)
-            : base(innerObject)
+        private ISqlResourcesOperations Client { get; set; }
+        private SqlDatabaseCreateUpdateParameters createUpdateParameters;
+        private ThroughputSettingsUpdateParameters ThroughputSettingsToUpdate;
+        private SqlContainersImpl sqlContainers;
+
+        internal SqlDatabaseImpl(string name, CosmosDBAccountImpl parent, SqlDatabaseGetResultsInner inner, ISqlResourcesOperations client)
+            : base(name, parent, inner)
+        {
+            this.Client = client;
+            this.sqlContainers = new SqlContainersImpl(this.Client, this);
+            SetCreateUpdateParameters();
+        }
+
+        private void SetCreateUpdateParameters()
         {
+            this.createUpdateParameters = new SqlDatabaseCreateUpdateParameters()
+            {
+                Location = Parent.Inner.Location,
+                Resource = new SqlDatabaseResource(id: this.Name()),
```

Comment:
Why `id = Name()`?

And it seems not compatible with 

```
        public string Id()
        {
            return this.Inner.Id;
        }
```

Reply:
This is what the swagger said. It is the id in resource of parameter, not the id in inner.